x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/api/util/tracing.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { Logger } from '@kbn/core/server';
+import { APMTracer } from '@kbn/langchain/server/tracers/apm';
+import { getLangSmithTracer } from '@kbn/langchain/server/tracers/langsmith';
+import type { Callbacks } from '@langchain/core/callbacks/manager';
+import type { LangSmithOptions } from '../../../../../../common/siem_migrations/model/common.gen';
+
+export const createTracersCallbacks = (
+  langsmithOptions: LangSmithOptions | undefined,
+  logger: Logger
+): Callbacks => {
+  const { api_key: apiKey, project_name: projectName = 'default' } = langsmithOptions ?? {};
```

Comment:
Does it makes sense to inherit `api_key` and `project_name` from langchain env variables ( in case these are not provided by the user) since they are already provided.

Reply:
Yes, it makes complete sense, and that's exactly what the langchanin library does internally when they are not provided in the parameters :)

Reply:
Hmm.. strange.... I tried to pass with blank `api_key` and i got the error that `api_key` is required so in that case we should make it optional, no?

![image](https://github.com/user-attachments/assets/579a669e-f1f2-45a5-adfa-3626a75727c2)

Reply:
Oh yeah, because they are required by our own API schema. 
We could turn them optional, but that would only make sense when running it locally. In a real server we need to force the client to provide their LangSmith API Key.